pkg/nodeset/v1alpha1/types.go

```diff
@@ -59,6 +61,9 @@ type NodeClassResource struct {
 	Reference *v1.ObjectReference `json:"reference,omitempty"`
 }
 
+// +genclient=true
+// +nonNamespaced=true
```

Comment:
Client-gen generats a non-namespaced version interface. Then update-codegen.sh script injects a fixed namespace into the client request calls.